### PR TITLE
Add a --wheel flag for generating `.whl` distributions.

### PR DESCRIPTION
**Pull request by pennomi**
Here's the first revision of the wheel generator, ready for criticism. I've only tested on Linux, but I don't see any reason it shouldn't work on all platforms as-is.

For Windows and Mac, we can upload these archives to PyPI, but for Linux, we'll need to host a URL for them to live at, since PyPI doesn't allow binary distributions for Linux.

### CODE REVIEW

File: makepanda/makewheel.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    if "linux" in p:
+        print(ColorText("red", "WARNING:") +
+              " Linux-specific wheel files are not supported."
+              " We will generate this wheel as a generic package instead.")
+        return "any"
+    return p
+
+
+# Global filepaths
+LIB_SRC = join("built", "lib")  # TODO: A special wheel lib dir
```
Comment (rdb) [inline makepanda/makewheel.py]: The "built" directory should be a parameter to the makewheel function.  On 64-bit Windows it is called "built_x64" and we actually allow overriding this (which I do a lot, and so does the buildbot) via the --outputdir parameter.

See makepandacore.GetOutputDir()

File: makepanda/makewheel.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    """
+    for f in os.listdir(src):
+        shutil.copy2(join(src, f), join(target, f))
+
+
+def makewheel(version):
+    # Global parameters that depend on version
+    data_target = join(WHEEL_TARGET, "panda3d-{}.data".format(version))
+    info_target = join(WHEEL_TARGET, "panda3d-{}.dist-info".format(version))
+    wheel_name = "panda3d-{}-{}-none-{}.whl".format(
```
Comment (rdb) [inline makepanda/makewheel.py]: abi-tag should not be "none" but "cp27", etc.

Comment (rdb) [inline makepanda/makewheel.py]: See https://www.python.org/dev/peps/pep-0425/ for more information about python tags and ABI tags - we need to indicate that we're fixed to CPython, and a particular version of it.

File: makepanda/makewheel.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+# Other global parameters
+PY_VERSION = "py3" if sys.version_info >= (3, 0) else "py2"
+PLATFORM_VERSION = get_platform()
+
+WHEEL_DATA = """Wheel-Version: 1.0
+Generator: bdist_wheel 1.0
+Root-Is-Purelib: false
+Tag: py2-none-any
+Tag: py3-none-any
```
Comment (rdb) [inline makepanda/makewheel.py]: Should this tag not match what is indicated by the filename, ie. dependence on a specific CPython version?

File: makepanda/makewheel.py (hunk 2 above)
Comment (pennomi) [inline makepanda/makewheel.py]: Note here that the wheel wouldn't install unless the ABI tag was "none". However, I did adjust the python version to explicitly require cpython.

**Comment by auvipy** [comment]
thats awesome move!!

File: makepanda/makewheel.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    whl.write_file_data(info_dir + '/metadata.json', json.dumps(METADATA, indent=4, separators=(',', ': ')))
+    whl.write_file_data(info_dir + '/METADATA', metadata)
+    whl.write_file_data(info_dir + '/WHEEL', WHEEL_DATA.format(PY_VERSION, ABI_TAG, PLATFORM_TAG))
+    whl.write_file(info_dir + '/LICENSE.txt', license_src)
+    whl.write_file(info_dir + '/README.md', readme_src)
+    whl.write_file_data(info_dir + '/top_level.txt', 'direct\npanda3d\npandac\n')
+
+    whl.close()
+
+# makewheel("1.10.0", "built_x64")
```
Comment (auvipy) [inline makepanda/makewheel.py]: add a new line here

**Comment by rdb** [comment]
OK, binary .whl packages for Panda3D 1.9.3 for Windows, macOS and Linux have been uploaded to PyPI.  `pip install panda3d` should now work on those platforms.  I'd welcome people to help test to make sure that every aspect of Panda3D still works correctly when installed via pip.

Thanks again!

**Comment by auvipy** [comment]
eagerly waiting for 1.10 stable release with py3k support <3

**Comment by pennomi** [comment]
I've done some very simple tests on Linux and can confirm it's working when installed via pip on Python 2. All of my bigger projects are on Python 3 however, so I can't test much more than a couple of tech demo programs.

**Comment by auvipy** [comment]
I guess pip3 install from github master branch should also work fine?

**Comment by pennomi** [comment]
Installing from source almost certainly won't work. Unlike most installable python packages, the wheel file is directly compiled into a binary format, bypassing the normal `setup.py` process.

Basically, what we're doing is very atypical for a python package, but is necessary due to the complex compilation process.

**Comment by wezu** [comment]
Tested the pip version on windows 10 and python 2.7.13. All the samples and bullet samples worked.

**Pull request closed without merging**